nixos/modules/services/hardware/keyd.nix

```diff
@@ -35,24 +32,64 @@ in
           };
         };
         description = lib.mdDoc ''
-          Configuration, except `ids` section, that is written to {file}`/etc/keyd/default.conf`.
+          Configuration, except `ids` section, that is written to {file}`/etc/keyd/<keyboard>.conf`.
+          Appropriate names can be used to write non-alpha keys, for example "equal" instead of "=" sign (see <https://github.com/NixOS/nixpkgs/issues/236622>).
           See <https://github.com/rvaiya/keyd> how to configure.
         '';
       };
     };
   };
+in
+{
+  options.services.keyd = {
+    enable = mkEnableOption (lib.mdDoc "keyd, a key remapping daemon");
+
+    keyboards = mkOption {
+      type = types.attrsOf (types.submodule keyboardOptions);
+      default = { };
+      example = literalExpression ''
+        {
+          default = {
+            ids = [ "*" ];
+            settings = {
+              main = {
+                capslock = "overload(control, esc)";
+              };
+            };
+          };
+          external_keyboard = {
+            ids = [ "1ea7:0907" ];
```

Comment:
Afaik, it will just read the configuration from the `default` file and the others will be ignored because `default` is like the number 1 file. I don't know all the configuration pitfalls, but people who need this feature (multiple files) know what they're doing (I hope)

Reply:
IIUC, the `external_keyboard` config has no effect.  As an example, it's misleading and should be fixed.

Reply:
No, it will have an effect in general, but not if you have `["*"]` id in both `default` and another file. Therefore `default.conf` has higher priority.
In this example, `externalKeyboard` is correct because it does not have an `*` identifier, but a concrete one.
Is this what you mean?